tests/cli-misc.rs

```diff
@@ -785,13 +786,26 @@ fn completion_cargo_unsupported_shell() {
 }
 
 #[test]
-fn set_profile() {
+fn set_profile_doesnt_change_installed_components() {
     setup(&|config| {
         expect_stdout_ok(config, &["rustup", "show", "profile"], "default");
-        assert!(config.exedir.join(format!("rls{}", EXE_SUFFIX)).exists());
+        expect_ok(
+            config,
+            &[
+                "rustup",
+                "toolchain",
+                "install",
+                "--no-self-update",
+                "stable",
+            ],
+        );
+        expect_component_not_executable(config, "rls");
         expect_ok(config, &["rustup", "set", "profile", "minimal"]);
         expect_stdout_ok(config, &["rustup", "show", "profile"], "minimal");
-        assert!(!config.exedir.join(format!("rls{}", EXE_SUFFIX)).exists());
+        expect_component_not_executable(config, "rls");
+        expect_ok(config, &["rustup", "set", "profile", "complete"]);
+        expect_stdout_ok(config, &["rustup", "show", "profile"], "complete");
+        expect_component_not_executable(config, "rls");
```

Comment:
Hmm, I wonder if this is the behaviour we should expect? Do you think we should install the missing components at the next `update`?

Reply:
I don't think we should alter the set of installed components when updating a toolchain, no.  The profile is essentially the default set of optional components to install when installing a fresh toolchain.  At least that's how I'm now thinking about it.